src/sentry/workflow_engine/endpoints/project_detector_details.py

```diff
@@ -74,6 +77,50 @@ def get(self, request: Request, project: Project, detector: Detector):
         )
         return Response(serialized_detector)
 
+    @extend_schema(
+        operation_id="Update a Detector",
+        parameters=[
+            GlobalParams.ORG_ID_OR_SLUG,
+            GlobalParams.PROJECT_ID_OR_SLUG,
+            DetectorParams.DETECTOR_ID,
+        ],
+        request=PolymorphicProxySerializer(
+            "GenericDetectorSerializer",
+            serializers=[
+                gt.detector_validator for gt in grouptype.registry.all() if gt.detector_validator
+            ],
+            resource_type_field_name=None,
+        ),
+        responses={
+            200: DetectorSerializer,
+            400: RESPONSE_BAD_REQUEST,
+            401: RESPONSE_UNAUTHORIZED,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOT_FOUND,
+        },
+    )
+    def put(self, request: Request, project: Project, detector: Detector) -> Response:
+        """
+        Update a Detector
+        ````````````````
+        Update an existing detector for a project.
+        """
+        group_type = None
+        detector_group_type = detector.group_type
+
+        if detector_group_type:
+            group_type = request.data.get("detector_type") or detector_group_type.slug
+
+        if not group_type:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
```

Comment:
this is all for annoying typing reasons - if `detector.group_type` is `None` we can't get the slug

Reply:
should we raise an exception if it's `None` in `Detector.group_type` so we can assume the return type is `GroupType`?